src/muxer.ts

```diff
@@ -10,18 +10,28 @@ export class DataChannelMuxerFactory implements StreamMuxerFactory {
    * WebRTC Peer Connection
    */
   private readonly peerConnection: RTCPeerConnection
+  private streamBuffer: WebRTCStream[] = []
 
-  /**
-   * The string representation of the protocol, required by `StreamMuxerFactory`
-   */
-  protocol: string = '/webrtc'
-
-  constructor (peerConnection: RTCPeerConnection) {
+  constructor (peerConnection: RTCPeerConnection, readonly protocol = '/webrtc') {
     this.peerConnection = peerConnection
+    // store any datachannels opened before upgrade has been completed
+    this.peerConnection.ondatachannel = ({ channel }) => {
+      const stream = new WebRTCStream({
+        channel,
+        stat: {
+          direction: 'inbound',
+          timeline: { open: 0 }
+        },
+        closeCb: (_stream) => {
+          this.streamBuffer = this.streamBuffer.filter(s => !_stream.eq(s))
+        }
+      })
+      this.streamBuffer.push(stream)
```

Comment:
There can be a maximum of 2^16 - 1 datachannels. This buffer would only hold references to objects. Please note that a PeerConnection could also initiate 2^16 datachannels after the handshake completes. Since the PeerConnection only notifies the user that a datachannel has been created by the remote using an `ondatachannel` callback.